### PR TITLE
Bug Fixes

### PR DESCRIPTION
**Pull request by clabraham2**
Connect writeRetract() and safePositionMethod property. Support clearanceHeight mode. (may wish to disable warning at 2927) Fix partsCounter211 key. (Either add 211 to key call, or remove 211 from property key)
setCurrentABC(abc) after G100 workplane command for machine simulation.

### CODE REVIEW

**Comment by ewildgoose** [comment]
That's great stuff, thanks for the contribution! Good catch on the silly mistake with the 211 counter, uff..

Can you explain why we need the "setCurrentABC()" call in that position? This is something that disappeared from the recent autodesk stuff I think? Also there are a couple of code paths that can reach that point, especially considering that we might be doing things like dogleg moves (or not) or avoiding that code path if we already changed tool (and another feature request is "safe starts", so we might be here in a redundant sense)? Thanks

**Comment by clabraham2** [comment]
setCurrentABC() is just for the machine simulation pose, no effect on output code. defineWorkPlane() is not invoked if useTiltedWorkplane is false, and therefore skipped before writeToolCall() which sets the rotary position with G100, but the machine simulation pose must still be updated for a 4 axis or 5 axis machine if not using tilted workplane. (i.e. programming from center of rotation)

**Comment by ewildgoose** [comment]
I've pulled in the partscounter fix and the changes to safe position height into the new "main" branch, which is also rebased onto latest autodesk (only a minor change)

I haven't (yet) integrated the setCurrentABC() because I don't quite understand why it's needed? I've fiddled with trying to get my machine model to simulate and that is itself a bit of a fail right now, however, I am not clear how to see the effects of this change?

It also seems to me that the same change could be added into an else statement in "setWorkPlane()"? What do you think?

**Comment by clabraham2** [comment]
tl;dr: COMMAND_LOAD_TOOL will output, in the nc code, the ABC angle in the G100 line without updating the machine simulation pose. setWorkPlane() is redundant, so setCurrentABC() is needed to update the machine sim pose.

There is a complicated thread of logic happening with defineWorkPlane(), setWorkPlane(), settings.workPlaneMethod.useTiltedWorkplane, and COMMAND_LOAD_TOOL. Putting setCurrentABC() where I did appeared to be the least invasive and does not double invoke defineWorkPlane/setWorkPlane unnecesarilly. setWorkPlane() is intended to _output_ the ABC positioning line in the posted code, however COMMAND_LOAD_TOOL retreives the ABC position via defineWorkPlane and outputs that position in the G100 line (instead of a seperate line, as defineWorkPlane/setWorkPlane would do). Without setCurrentABC() in COMMAND_LOAD_TOOL, the workplane is calculated without outputting the position on its own line (which is correct), ABC angle is then set by G100, but the machine simulation pose is not updated.

If settings.workPlaneMethod.useTiltedWorkplane is true, then the positioning of the rotary axes happens after COMMAND_LOAD_TOOL (i.e. after G100) so that the angles are utilized with G68.2.

**Comment by ewildgoose** [comment]
Implemented in latest master branch. Thanks for the feedback!
